lib/BlockedBy.js

```diff
@@ -103,6 +103,41 @@ async function blockedBySelf({ waitingKey, buildId, collapse }) {
 
     return false;
 }
+
+/**
+ *
+ * @param {String} buildConfig
+ * @param {Array} blockingBuildIds
+ * @param {String} buildId
+ * @param {Object} redisConn
+ */
+async function checkMultipleBuildsFromSameEvent(buildConfig, blockingBuildIds, buildId, redisConn) {
+    if (!buildConfig) return false;
+
+    const parsedConfig1 = JSON.parse(buildConfig);
+
+    if (!parsedConfig1 || !parsedConfig1.eventId) {
+        return false;
+    }
+
+    const isSameEvent = await Promise.race(blockingBuildIds.map(async (b) => {
+        const bConfigs = await redisConn.hget(
+            `${queuePrefix}buildConfigs`, b
+        );
+        const parsedConfig2 = JSON.parse(bConfigs);
+        const hasEventId = parsedConfig2 && parsedConfig1.eventId;
+        const compare = hasEventId && (parsedConfig1.eventId === parsedConfig2.eventId);
+
+        if (compare) {
+            logger.info(`Builds ${b} & ${buildId} have the same event ${parsedConfig1.eventId}`);
```

Comment:
let's do `log.error` here, and log eventId & both buildIds, also should setup a cloudwatch alarm on this log.